java-checks/src/test/files/checks/AnonymousClassShouldBeLambdaCheck_no_version.java

```diff
@@ -92,10 +106,90 @@ public String handle() {
         return A.this.toStr();
       }
     };
+   
+    new B() {// Compliant
+      @Override
+      public void bar() {
+      }
+    };
+  
+    new C() { // Noncompliant
+      @Override
+      public void foo() {
+      }
+    };
+   
+    new D() {};  // Compliant
+  
+    new D() { // Noncompliant
+      @Override
+      public void foo2() {}
+    };
   }
 
   String toStr(){
     return "";
   }
 
+  interface A2 {
+    default void bar1() {}
+    default void foo() {}
+  }
+  
+  interface A1 extends A2{
+    default void foo(int a) {}
+  }
+
+  interface B1 extends A1 {
+    default void bar() {}
+    default String bar4(int a, int b) {}
+    default Integer bar2(int a, int b, String c) {}
+  }
+
+  class C1 {
+    B1 b1 = new B1() { // Compliant
+      @Override
+      public void bar() {
+        System.out.println();
+      }
+    };
+  }
+
+  class C2 {
+    B1 b1 = new B1() { // Compliant
+      @Override
+      public void foo() {
+        System.out.println();
+      }
+    };
+  }
+
+  interface F extends B1{
+    default String bar(int a, int b) {}
```

Comment:
does not compile, missing return statement, same error for ALL your non-void default methods.